packages/actor-query-operation-quadpattern/lib/ActorQueryOperationQuadpattern.ts

```diff
@@ -49,12 +87,47 @@ export class ActorQueryOperationQuadpattern extends ActorQueryOperationTyped<Alg
 
   public async runOperation(pattern: Algebra.Pattern, context?: {[id: string]: any})
   : Promise<IActorQueryOperationOutputBindings> {
-    // Resolve the quad pattern
-    const result = await this.mediatorResolveQuadPattern.mediate({ pattern, context });
-
     // Collect all variables from the pattern
     const variables: string[] = this.getVariables(pattern);
 
+    let quadPattern: RDF.Quad = pattern;
+    let bindGraph: RDF.Term = null;
+    // Check if the available named graphs were specified
+    if (context && context.namedGraphs) {
+      if (!context.namedGraphs.length) {
+        // When no named graphs were deliberatively selected, we can't query anything
+        return ActorQueryOperationQuadpattern.newEmptyResult(variables);
```

Comment:
why can't the default graph be queried in this case? (It is not part of the namedGraphs array as far as I can see)

Reply:
This situation can actually never occur for SPARQL queries. If no FROM NAMEDs are in the query, `namedGraphs` will be undefined, not empty.

I thought it might be nice to have this case (defined but empty `namedGraphs`) internally for flexibility. 

But I can remove it if you think it doesn't make sense :-)

Reply:
Ok, I'll just remove that part to avoid confusion.

You're right about the second thing, I'll fix that.